common/transport/amqp/src/sender_link.ts

```diff
@@ -0,0 +1,207 @@
+import * as machina from 'machina';
+import * as amqp10 from 'amqp10';
+import * as dbg from 'debug';
+import { EventEmitter } from 'events';
+import { Message, results } from 'azure-iot-common';
+import { AmqpMessage } from './amqp_message';
+import { AmqpLink } from './amqp_link_interface';
+
+const debug = dbg('SenderLink');
+
+interface MessageOperation {
+  message: Message;
+  callback: (err?: Error, result?: results.MessageEnqueued) => void;
+}
+
+/*Codes_SRS_NODE_AMQP_SENDER_LINK_16_002: [The `SenderLink` class shall inherit from `EventEmitter`.]*/
+/*Codes_SRS_NODE_AMQP_SENDER_LINK_16_003: [The `SenderLink` class shall implement the `AmqpLink` interface.]*/
+export class SenderLink extends EventEmitter implements AmqpLink {
+  private _linkAddress: string;
+  private _linkOptions: any;
+  private _linkObject: any;
+  private _fsm: machina.Fsm;
+  private _amqp10Client: amqp10.AmqpClient;
+  private _unsentMessageQueue: MessageOperation[];
+  private _pendingMessageQueue: MessageOperation[];
+  private _detachHandler: (detachEvent: any) => void;
+  private _errorHandler: (err: Error) => void;
+
+  constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient) {
```

Comment:
>constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient) { [](start = 2, length = 85)

How do I create a sender link such that any messages I send over THAT link are send pre-settled?